primitives/src/conststr.rs

```diff
@@ -271,6 +270,33 @@ impl WrapperTypeDecode for ConstString {
     type Wrapped = String;
 }
 
+// darling doesn't support unions, so this can't be derived
+impl TypeId for ConstString {
+    fn id() -> Ident {
+        #[cfg(not(feature = "std"))]
+        use alloc::string::ToString;
+        "ConstString".to_string()
+    }
+}
+impl IntoSchema for ConstString {
+    fn type_name() -> Ident {
+        #[cfg(not(feature = "std"))]
+        use alloc::string::ToString;
+        "String".to_string()
```

Comment:
Shouldn't it be `ConstString`?

Reply:
Or better just reuse implementation from `TypeId`

Reply:
This is how the derive works: it's a transparent type so it reports itself as a `String`. For `TypeId`, however, transparent types do not matter and it reports the actual type name